rhel/entrypoint.sh

```diff
@@ -1,22 +1,87 @@
 #!/bin/bash -x
-mount_rootfs() {
-    echo "Mounting OFED driver container rootfs..."
+
+function set_driver_readiness() {
+    touch /.driver-ready
+}
+
+function unset_driver_readiness() {
+    rm -f /.driver-ready
+}
+
+function exit_on_error() {
+    $@
+    if [[ $? -ne 0 ]]; then
+        echo "Error occured while executing: $1"
+        exit 1
+    fi
+}
+
+function mount_rootfs() {
+    echo "Mounting Mellanox OFED driver container rootfs..."
     mount --make-runbindable /sys
     mount --make-private /sys
     mkdir -p /run/mellanox/drivers
     mount --rbind / /run/mellanox/drivers
 }
 
-unmount_rootfs() {
+function unmount_rootfs() {
     echo "Unmounting Mellanox OFED driver rootfs..."
     if findmnt -r -o TARGET | grep "/run/mellanox/drivers" > /dev/null; then
       umount -l -R /run/mellanox/drivers
     fi
 }
 
-/etc/init.d/openibd restart
-ofed_info -s
+function handle_signal() {
+    unset_driver_readiness
+    unmount_rootfs
+    echo "Stopping Mellanox OFED Driver..."
+    /etc/init.d/openibd force-stop
+    exit 0
+}
+
+function ofed_exist_for_kernel() {
+    # check if mlx5_core exists in dkms under running kernel, this should be sufficient to hint us if
+    # OFED drivers are installed for the running kernel
+    local KVER=$(uname -r)
+    if [[ -e /usr/lib/modules/${KVER}/extra/mlnx-ofa_kernel/drivers/net/ethernet/mellanox/mlx5/core/mlx5_core.ko ]]; then
+        echo "OFED driver found for kernel"
+        return 0
+    fi
+    echo "No OFED driver found for kernel ${KVER}"
+    return 1
+
+}
+
```

Comment:
@adrianchiris You mentioned that the driver may work with weak-updates, there is a tool in the kmod RPM package that checks the kernel module against the running kernel and does the symlinking if the modules are compatible (`weak-modules`). Before rebuilding we should check if we can load the modules after the run of `weak-modules` and then do a rebuild. /cc @sjug

Reply:
yes so it would be:

1. see if driver is compatible with running -> if yes load driver
2. attempt weak updates -> attempt to load else goto 3
3. rebuild - load driver

right ?